app/models/infrastructure_facades/pl_grid_facade.rb

```diff
@@ -140,27 +138,24 @@ def self.remove_local_monitoring_config(sm_uuid)
   end
 
   def self.start_monitoring_cmd
-    chain(
-        cd(RemoteDir::scalarm_root),
-        "unxz -f #{ScalarmFileName::monitoring_package}",
-        "chmod a+x #{ScalarmFileName::monitoring_binary}",
-        "export X509_USER_PROXY=#{ScalarmFileName::remote_proxy}",
-        "#{run_in_background("./#{ScalarmFileName::monitoring_binary} #{ScalarmFileName::monitoring_config}",
-                                           "#{ScalarmFileName::monitoring_binary}_`date +%Y-%m-%d_%H-%M-%S-$(expr $(date +%N) / 1000000)
-`.log")}"
-    )
+    BashCommand.new.cd(RemoteDir::scalarm_root).
+        append("unxz -f #{ScalarmFileName::monitoring_package}").
+        append("chmod a+x #{ScalarmFileName::monitoring_binary}").
+        append("export X509_USER_PROXY=#{ScalarmFileName::remote_proxy}").
+        run_in_background("./#{ScalarmFileName::monitoring_binary} #{ScalarmFileName::monitoring_config}",
+          "#{ScalarmFileName::monitoring_binary}_`date +%Y-%m-%d_%H-%M-%S-$(expr $(date +%N) / 1000000))`.log").to_s
   end
 
   def self.clone_proxy(ssh, remote_path)
     # TODO: checking if proxy file exists?
-    ssh.exec! "cp `voms-proxy-info -p` #{remote_path}"
+    ssh.exec! BashCommand.new.append("cp `voms-proxy-info -p` #{remote_path}").to_s
```

Comment:
Z tego co udalo mi sie znalesc w kodzie to jest jedno cp. Nie ma wiec sensu dodawac tej metody.